src/core/server/http/http_server.test.ts

```diff
@@ -68,7 +68,11 @@ beforeEach(() => {
     port: 10002,
     ssl: { enabled: false },
     compression: { enabled: true },
-  } as HttpConfig;
+    requestOpaqueId: {
+      allowFromAnyIp: true,
+      ipAllowlist: [],
+    },
+  } as any;
```

Comment:
Is this header always going to be X-Opaque-Id or do we potentially need to support extracting the request id from a different header?

Minor suggestion would be to align the naming of this config option with the `request.id` property of `KibanaRequest` object. (Either name this config option `settings.requestId` or the property `request.opaqueId`

Reply:
We definitely may want to make that header configurable in the future. Agree that we should make the config key name more generic then. Let's go with `server.requestId` 👍 I will update the PR shortly.